airflow/models/taskinstance.py

```diff
@@ -1518,7 +1518,8 @@ def _run_raw_task(
         if not test_mode:
             session.add(Log(self.state, self))
             session.merge(self)
-            self._create_dataset_dag_run_queue_records(session=session)
+            if self.state != State.SKIPPED:
```

Comment:
Wait, should we test if it's in State.SUCCESS instead?

Reply:
Also, you should be using `TaskInstanceState` instead of `State`.

Reply:
> Wait, should we test if it's in State.SUCCESS instead?

i think the only way you get here is if you are success or skipped but yeah probably best to just check for success 👍 

when i first did this i thought you could only get here if _successful_ but missed the skipped part